aerich/migrate.py

```diff
@@ -189,21 +190,19 @@ def _add_operator(cls, operator: str, upgrade=True, fk_m2m_index=False) -> None:
 
     @classmethod
     def _handle_indexes(cls, model: Type[Model], indexes: List[Union[Tuple[str], Index]]) -> list:
-        ret: list = []
+        if index_classes := set(index.__class__ for index in indexes if isinstance(index, Index)):
+            for index_cls in index_classes:
+                if index_cls(fields=("id",)) != index_cls(fields=("id",)):
 
-        def index_hash(self) -> str:
-            h = hashlib.new("MD5", usedforsecurity=False)  # type:ignore[call-arg]
-            h.update(
-                self.index_name(cls.ddl.schema_generator, model).encode()
-                + self.__class__.__name__.encode()
-            )
-            return h.hexdigest()
+                    def _hash(self) -> int:
+                        return hash((tuple(sorted(self.fields)), self.name, self.expressions))
 
-        for index in indexes:
-            if isinstance(index, Index):
-                index.__hash__ = index_hash  # type:ignore[method-assign,assignment]
-            ret.append(index)
-        return ret
+                    def _eq(self, other) -> bool:
+                        return type(self) is type(other) and hash(self) == hash(other)
+
+                    setattr(index_cls, "__hash__", _hash)
+                    setattr(index_cls, "__eq__", _eq)
```

Comment:
You are setting the lower boundary on the tortoise version in https://github.com/tortoise/aerich/pull/388. Why not just implement `__eq__` and `__hash__` in Tortoise and set the corresponding version boundary. You won't have to have this hacky code then.

Reply:
@henadzit I check the tortoise version before patch Index and add a `TODO` mark.

Reply:
Hey, can you please let me know your arguments for why you want to have the patching code instead of increasing the version requirements?